tests/core/test_datamodules.py

```diff
@@ -434,9 +434,9 @@ def train_dataloader(self):
         return DataLoader(self.mnist_train, batch_size=4)
 
 
-def test_dm_reload_dataloaders_every_epoch(tmpdir):
+def test_dm_reload_dataloaders_every_n_epochs(tmpdir):
```

Comment:
is this test even doing anything? looks similar to what we have in `trainer/test_dataloaders.py`. I see a few more tests being duplicated in tests/datamodules/ which are already present in tests/trainer/.

Reply:
looks like this test was originally made to check if custom datamodules work with  reload_every_epoch 

this test fits the model with a `CustomMNISTDataModule` instead of the using the dataloaders in `EvalModelTemplate` as the other tests in `trainer/test_dataloaders.py` do. if there's no difference here then `test_dataloaders_load_every_n_epochs` should cover this test

Reply:
I'd keep it. Just update the docstring with more info about what the the test actually tests because currently it seems redundant.

Context here: https://github.com/PyTorchLightning/pytorch-lightning/issues/3974#issuecomment-705332476

Reply:
I'd keep it. Just update the docstring with more info about what the the test actually tests because as it is, seems redundant.

Context here: https://github.com/PyTorchLightning/pytorch-lightning/issues/3974#issuecomment-705332476